core/aws-core/src/main/java/software/amazon/awssdk/awscore/eventstream/EventStreamAsyncResponseTransformer.java

```diff
@@ -205,8 +226,18 @@ private MessageDecoder createDecoder() {
                                                                             EMPTY_EXECUTION_ATTRIBUTES));
                     }
                 } else if (isError(m) || isException(m)) {
-                    Throwable exception = exceptionUnmarshaller.handle(adaptMessageToResponse(m, true),
-                                                                       EMPTY_EXECUTION_ATTRIBUTES);
+                    SdkHttpFullResponse errorResponse = adaptMessageToResponse(m, true);
+                    if (!errorResponse.firstMatchingHeader(X_AMZN_REQUEST_ID_HEADER).isPresent() &&
```

Comment:
I think you can omit the first part of the conditional. The error response shouldn't have a request id.

Reply:
ok, I was thinking in case the format changes in future. Will update